src/redux/uniswap.ts

```diff
@@ -0,0 +1,224 @@
+const UNISWAP_CLEAR_STATE = 'uniswap/UNISWAP_CLEAR_STATE';
+
+// -- Actions --------------------------------------------------------------- //
+
+/**
+ * Represents the current state of the `uniswap` reducer.
+ */
+interface UniswapState {
+  /**
+   * An array of addresses for the user's favorite Uniswap pairs.
+   */
+  favorites: string[];
+
+  /**
+   * Whether or not data from Uniswap is currently being loaded.
+   */
+  loadingUniswap: boolean;
+
+  /**
+   * Data for Uniswap pairs as an object mapping addresses to tokens.
+   */
+  pairs: Record<string, RainbowToken>;
+}
+
+/**
+ * An action for the `uniswap` reducer.
+ */
+type UniswapAction =
+  | UniswapLoadRequestAction
+  | UniswapLoadSuccessAction
+  | UniswapLoadFailureAction
+  | UniswapUpdatePairsAction
+  | UniswapUpdateFavoritesAction
+  | UniswapClearStateAction;
+
+/**
+ * The action for starting to load data from Uniswap.
+ */
+interface UniswapLoadRequestAction {
+  type: typeof UNISWAP_LOAD_REQUEST;
+}
+
+/**
+ * The action used when data from Uniswap is loaded successfully.
+ */
+interface UniswapLoadSuccessAction {
+  type: typeof UNISWAP_LOAD_SUCCESS;
+  payload: UniswapState['favorites'];
+}
+
+/**
+ * The action used when loading data from Uniswap fails.
+ */
+interface UniswapLoadFailureAction {
+  type: typeof UNISWAP_LOAD_FAILURE;
+}
+
+/**
+ * The action for updating Uniswap pair data.
+ */
+interface UniswapUpdatePairsAction {
+  type: typeof UNISWAP_UPDATE_PAIRS;
+  payload: UniswapState['pairs'];
+}
+
+/**
+ * The action for updating Uniswap favorites.
+ */
+interface UniswapUpdateFavoritesAction {
+  type: typeof UNISWAP_UPDATE_FAVORITES;
+  payload: UniswapState['favorites'];
+}
+
+/**
+ * The action for resetting the state.
+ */
+interface UniswapClearStateAction {
+  type: typeof UNISWAP_CLEAR_STATE;
+}
+
+/**
+ * Loads Uniswap favorites from global storage and updates state.
+ */
+export const uniswapLoadState = () => async (
+  dispatch: Dispatch<
+    | UniswapLoadRequestAction
+    | UniswapLoadSuccessAction
+    | UniswapLoadFailureAction
+  >,
+  getState: AppGetState
+) => {
+  const { network } = getState().settings;
+  dispatch({ type: UNISWAP_LOAD_REQUEST });
+  try {
+    const favorites: string[] = await getUniswapFavorites(network);
+    remove(favorites, address => toLower(address) === toLower(SOCKS_ADDRESS));
+    dispatch({
+      payload: favorites,
+      type: UNISWAP_LOAD_SUCCESS,
+    });
+  } catch (error) {
+    dispatch({ type: UNISWAP_LOAD_FAILURE });
+  }
+};
+
+/**
+ * Updates state to use initial data for Uniswap pairs.
+ */
+export const uniswapPairsInit = () => (
+  dispatch: Dispatch<UniswapUpdatePairsAction>,
+  getState: AppGetState
+) => {
+  const { network } = getState().settings;
+  const pairs =
+    network === networkTypes.mainnet
+      ? rainbowTokenList.CURATED_TOKENS
+      : getTestnetUniswapPairs(network);
+  dispatch({
+    // @ts-ignore
+    payload: pairs,
```

Comment:
`getTestnetUniswapPairs` returns `Asset`s as opposed to `RainbowToken`s. `RainbowToken` is assumed elsewhere in places that use this reducer, so I kept it as the pair type. I'm not sure of the semantic difference between those two types regardless though, so it might make sense to do `RainbowToken | Asset` for the pairs?

Reply:
the ts-expect-error is good here! we'll need to consolidate our "uniswap pairs" assets - but first we will be moving over to our new swap aggregator, so we will want to define our new data then!